capstone/src/main/webapp/scripts/group.js

```diff
@@ -1,21 +1,94 @@
 let groupId;
+let locationsLimit = 1;
 
 function init() {
+  createAPIKey();
   getGroupId();
   checkMembership();
   createLogoutUrl();
   loadPosts();
   getPollOptions();
   fetchBlobstoreUrlAndShowForm();
   loadMembers();
+  findClosestGroupLocations();
   autocomplete();
   loadTags();
 }
 
+function createAPIKey() {
+  var xmlhttp = new XMLHttpRequest();
+
+  xmlhttp.onreadystatechange = function() {
+    if (this.readyState == 4 && this.status == 200) {
+      var myArr = JSON.parse(this.responseText);
+      let script = document.createElement('script');
+      script.src = "https://maps.googleapis.com/maps/api/js?key=" + myArr[0].GOOGLE_MAPS_API_KEY;
+      document.getElementsByTagName('head')[0].appendChild(script);
+    }
+  };
+  xmlhttp.open("GET", "scripts/api_key.json", true);
+  xmlhttp.send();
+}
+
 function getGroupId() {
   groupId = window.location.search.substring(1).split("groupId=")[1];
 }
 
+function findClosestGroupLocations() {
+  fetch(`/create-quadtree?groupId=${groupId}`, { method: "POST" });
+  fetch(`/central-group-locations?groupId=${groupId}`).then(response => response.json()).then((allCentralGroupLocations) => {
+    const groupLocations = document.getElementById("locations-container");
+    groupLocations.innerHTML = "";
+    for (let i = 1; i < locationsLimit + 1; i++) {
+      groupLocations.appendChild(createLocationComponent(allCentralGroupLocations[i]));
+    }
+    createMap(allCentralGroupLocations, locationsLimit);
+  });
+}
+
+function locationAmount() {
+  const amount = document.getElementById("number");
+  const value = amount.value;
+  if (value === "1") {
+    locationsLimit = 1;
+  } else if (value === "5") {
+    locationsLimit = 5;
+  } else if (value === "10") {
+    locationsLimit = 10;
+  } else if(value === "15") {
+    locationsLimit = 15;
+  } else if(value === "20") {
+    locationsLimit = 20;
+  } 
```

Comment:
You can use parseInt() to convert these to numbers instead so you don't if statements for each value you want